dts/arm/nordic/nrf51822.dtsi

```diff
@@ -92,7 +89,6 @@
 			clock-frequency = <I2C_BITRATE_STANDARD>;
 			interrupts = <3 NRF_DEFAULT_IRQ_PRIORITY>;
 			status = "disabled";
-			label = "I2C_0";
```

Comment:
Don't we need to still keep some of these labels for shell? At least `I2C_*` ones seem to be needed: https://github.com/zephyrproject-rtos/zephyr/blob/3b5590a18790ea8f9159eadbc38f8a514b342543/drivers/i2c/i2c_shell.c#L280

Reply:
So looking at this, I think we can rework this based on infrastructure we have for userspace.  We can find a list of devices that support the API of interest.

Reply:
Here's the starts of a change that implements a better filter/lookup mechanism:
https://github.com/zephyrproject-rtos/zephyr/pull/48381